docs/reference/concepts/05-events.md

```diff
@@ -0,0 +1,83 @@
+---
+sidebar_position: 1
+id: events
+---
+
+# Events
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+Events allow enable the ability to react to state changes in the provider or underlying flag management system.
+These include changes in provider readiness, error status, or perhaps most interestingly, flag configuration changes.
+
+## Event Handlers
+
+Event handlers can be attached to a `client` or to the global API object, for any [provider event type](/specification/types#provider-events).
+Handlers attached to the global API object run when _any_ provider emits the associated events; this makes global handlers ideal for configuration troubleshooting, monitoring and other cross-cutting concerns.
+Handlers attached to a client will run only when the provider bound with that client emits the associated event; this makes `client` handlers the ideal for reacting to flag-state changes in an application domain logic.
+
+Handlers are passed an [events details](/specification/types#event-details) structure, which contains data about the event, including a list of keys that have changed (if applicable and available).
+
+<!-- TODO: add more languages when completed -->
+<Tabs groupId="code">
+<TabItem value="js" label="TypeScript">
+
+```ts
+// attach a `PROVIDER_READY` handler
+client.addHandler(ProviderEvents.Ready, () => {
+  // do something when the provider is ready
+});
+
+// attach a `PROVIDER_CONFIGURATION_CHANGED` handler
+client.addHandler(ProviderEvents.ConfigurationChanged, () => {
+  // do something when flags settings have changed
+});
+
+// attach a `PROVIDER_ERROR` handler
+client.addHandler(ProviderEvents.Error, () => {
+  // do something when the provider has entered an error state
+});
+
+// attach a `PROVIDER_STALE` handler
+client.addHandler(ProviderEvents.Stale, () => {
+  // do something when the provider has gone stale
+});
+```
+
+</TabItem>
+</Tabs>
+
+## Event Types
+
+See [event types](/specification/types#provider-events) specification.
+
+### PROVIDER_READY
+
+The provider is ready to perform flag evaluations.
+
+Authors may not wish to attempt to evaluate flags until the provider has properly established a connection to the management system.
+In this case, authors can await the `PROVIDER_READY` event before evaluating any flags.
+This can be especially useful for hiding views or otherwise deferring evaluation until associated flags can be resolved accurately.
```

Comment:
What do you mean by "hiding views"?

Reply:
Hiding UI components, for example, showing  a loader.

Reply:
Or "hiding UI elements". Something like that would clear things up for me.

Reply:
I like that.